File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        #use a hashmap to track seen
        seen = {}
        for idx, value in enumerate(nums):
            if target - value in seen:
                return [idx,seen[target-value]]
            else:
                seen[value] = idx
        

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  let seen = {};
  //hashmap for seen values
  //iterate through nums array
  for (let i = 0; i < nums.length; i++) {
    if (target - nums[i] in seen) {
      return [seen[target - nums[i]], i] 
    } else {
      seen[nums[i]] = i  
    }
  }
    //if target - current number is in hashmap
      //return index value + current index
};